File: app/src/main/java/mydebts/android/app/MyDebtsApplicationComponent.java
package mydebts.android.app;

import javax.inject.Singleton;

import dagger.Component;
import mydebts.android.app.data.DataModule;
import mydebts.android.app.data.db.DbModule;

@Singleton
@Component(modules = {
        MyDebtsApplicationModule.class,
        DataModule.class,
        MyDebtsApplicationBindingModule.class})
interface MyDebtsApplicationComponent {

    void inject(MyDebtsApplication application);
}

File: app/src/main/java/mydebts/android/app/MyDebtsApplicationComponent.kt
package mydebts.android.app

import javax.inject.Singleton

import dagger.Component
import mydebts.android.app.data.DataModule

@Singleton
@Component(modules = arrayOf(MyDebtsApplicationModule::class, DataModule::class, MyDebtsApplicationBindingModule::class))
internal interface MyDebtsApplicationComponent {
    fun inject(application: MyDebtsApplication)
}
